src/main/java/org/valdroz/vscript/Variant.java

```diff
@@ -325,6 +325,45 @@ public String toString() {
         }
     }
 
+    /**
+     * Rounds the given value to the specified number of decimal places, following standard rounding rules.
+     * Handles nulls, non-numeric, negatives, and very large numbers gracefully.
+     *
+     * @param value Variant or numeric value to round (expressions supported)
+     * @param decimalPlaces number of decimal places to round to
+     * @return Variant containing rounded value (int if decimalPlaces==0, else float)
+     */
+    public static Variant round(Variant value, int decimalPlaces) {
+
+        if (decimalPlaces < 0) {
+            throw new EvaluationException("Invalid decimal places, must be a positive number: " + decimalPlaces);
+        }
+
+        if (value == null) {
+            return nullVariant();
+        }
+
+        BigDecimal num = value.asNumeric();
+
+        if (num == null) {
+            try {
+                num = new BigDecimal(value.toString());
```

Comment:
This need to be revisited. If `value.asNumeric()` returns `null` then `new BigDecimal(value.toString())` is irrelevant. What we wan is, if `num` value is null then consider it as zero.